### DISCUSSION TITLE
Add auto-updater

### DISCUSSION THREAD

**Discussion by HimanshuBansal112** [project64/project64#2263] [Ideas]
Project 64 has no auto-updater and it has been quite long time after its release so please add it.

**Comment by project64** [comment]
how do you see this working, is this meant to update on every new change to github, on major releases?

**Comment by oddMLan** [comment]
You pick an update channel. The final builds defaulting to the Release channel. Dev builds defaulting to Nightly. The user can switch from Release -> Dev but not the other way around (because new changes might break old files).

The updater has to handle special cases like [Settings] in .cfg changing to [default] and so on when these changes occur.

When a new update is out the user is prompted with a changelog and so can decide whether it's worth to update or not.

**Comment by HimanshuBansal112** [comment]
I have got two or three options in other emulators. Generally, there are-: stable , beta and dev update options.

**Comment by dftf-stu** [comment]
_DuckStation_ currently does this -- and is also hosted [here](https://github.com/stenzek/duckstation) on GitHub.

It simply pulls-down the latest ZIP file for the relevant channel you are on (stable or dev), and saves it inside the folder containing the app's files.  Then a separate updater app launches, kills any existing _DuckStation_ processes, un-zips the file and replaces any existing files of the same name, then re-loads the main _DuckStation_ interface while terminating itself.

Go to their project's [_Releases_ section](https://github.com/stenzek/duckstation/releases), and download the ZIP of the "Latest Development Build" and extract it somewhere.  Then, once you're in the app, go to Settings then General and change the "latest" dropdown, next to "Update channel:", to "preview".  Then close the app.  Next time you open it, it will trigger the update process, and you can see how they do it.

Would be great for Project64 to be able to do something similar!
